pinot-common/src/main/java/org/apache/pinot/common/messages/SegmentReloadMessage.java

```diff
@@ -34,19 +35,25 @@ public class SegmentReloadMessage extends Message {
   public static final String RELOAD_SEGMENT_MSG_SUB_TYPE = "RELOAD_SEGMENT";
 
   private static final String FORCE_DOWNLOAD_KEY = "forceDownload";
+  private static final String SEGMENT_LIST_KEY = "segmentList";
 
-  public SegmentReloadMessage(@Nonnull String tableNameWithType, @Nullable String segmentName, boolean forceDownload) {
+  public SegmentReloadMessage(@Nonnull String tableNameWithType, @Nullable List<String> segmentNames,
+      boolean forceDownload) {
     super(MessageType.USER_DEFINE_MSG, UUID.randomUUID().toString());
     setResourceName(tableNameWithType);
-    if (segmentName != null) {
-      setPartitionName(segmentName);
-    }
     setMsgSubType(RELOAD_SEGMENT_MSG_SUB_TYPE);
     // Give it infinite time to process the message, as long as session is alive
     setExecutionTimeout(-1);
 
     ZNRecord znRecord = getRecord();
     znRecord.setBooleanField(FORCE_DOWNLOAD_KEY, forceDownload);
+    if (segmentNames != null) {
+      if (segmentNames.size() == 1) {
+        setPartitionName(segmentNames.get(0));
+      } else {
+        znRecord.setListField(SEGMENT_LIST_KEY, segmentNames);
```

Comment:
Let's always set the `SEGMENT_LIST_KEY` regardless of the number of segments. In the future we can always read the list field. We need to also set the partition name for single segment for backward compatibility. Let's also add a TODO to remove it after the next release